src/Symfony/Component/Console/Messenger/ExecuteCommandHandler.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Messenger;
+
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Input\StringInput;
+use Symfony\Component\Console\Output\BufferedOutput;
+
+/**
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+final class ExecuteCommandHandler
+{
+    public function __construct(private readonly Application $application)
+    {
+    }
+
+    public function __invoke(ExecuteCommand $message): BufferedOutput
+    {
+        $input = new StringInput($message->input);
+        $output = new BufferedOutput();
+
+        $this->application->setCatchExceptions($message->catchExceptions);
+        $this->application->run($input, $output);
```

Comment:
Should we throw an exception if the exist code isn't `0`?

Reply:
perhaps only if message.catchExceptions=true we should try-catch Application.run() and log any exception to console channel vs. letting the Application exception bubble otherwise

about the exit code; Process components calls it "mustRun"

Reply:
I've made this configurable.